src/main/java/de/tum/in/www1/exerciseapp/domain/MultipleChoiceQuestionStatistic.java

```diff
@@ -0,0 +1,214 @@
+        this.answerCounters = answerCounters;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        MultipleChoiceQuestionStatistic multipleChoiceQuestionStatistic = (MultipleChoiceQuestionStatistic) o;
+        if (multipleChoiceQuestionStatistic.getId() == null || getId() == null) {
+            return false;
+        }
+        return Objects.equals(getId(), multipleChoiceQuestionStatistic.getId());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(getId());
+    }
+
+    @Override
+    public String toString() {
+        return "MultipleChoiceQuestionStatistic{" +
+            "id=" + getId() +
+            "}";
+    }
+
+    /**
+     * 1. creates the AnswerCounter for the new AnswerOption
+     *          if where is already an AnswerCounter with the given answerOption -> nothing happens
+     *
+     * @param answer the answer object which will be added to the MultipleChoiceStatistic
+     */
+    public void addAnswerOption(AnswerOption answer) {
+
+        if(answer ==null) {
+            return;
+        }
+
+        for(AnswerCounter counter: answerCounters) {
+            if(answer.equals(counter.getAnswer())) {
+                return;
+            }
+        }
+        AnswerCounter answerCounter = new AnswerCounter();
+        answerCounter.setAnswer(answer);
+        addAnswerCounters(answerCounter);
+
+    }
+
+    /**
+     * 1. check if the Result is rated or unrated
+     * 2. increase participants, all selected AnswerCounter and if the question is correct, than increase the correctCounter
+     *
+     * @param submittedAnswer the submittedAnswer object which contains all selected answers
+     * @param rated specify if the Result was rated ( participated during the releaseDate and the dueDate of the quizExercise)
+     *                                  or unrated  ( participated after the dueDate of the quizExercise)
+     */
+    @Override
+    public void addResult(SubmittedAnswer submittedAnswer, boolean rated) {
+
+        if(submittedAnswer == null) {
+            return;
+        }
+
+        MultipleChoiceSubmittedAnswer mcSubmittedAnswer = (MultipleChoiceSubmittedAnswer)submittedAnswer;
+
+        if(rated) {
+            //increase participants
+            setParticipantsRated(getParticipantsRated()+1);
+
+            if(mcSubmittedAnswer.getSelectedOptions() != null) {
+                // increase answerCounter if answer is selected
+                for (AnswerCounter answerCounter: answerCounters) {
+                    if(mcSubmittedAnswer.getSelectedOptions().contains(answerCounter.getAnswer())) {
+                        answerCounter.setRatedCounter(answerCounter.getRatedCounter()+1);
+                    }
+                }
+            }
```

Comment:
this code looks similar as the one below. Please avoid duplicating code and try to refactor (extract) it into another method

Reply:
this is not exactly the same code, the difference are the rated or unrated Counters, so I can't see a way to extract them into another method. (this comment is also valid for the other comments)